www/install/php/Update-21.10.10.php

```diff
@@ -0,0 +1,102 @@
+ * limitations under the License.
+ *
+ * For more information : contact@centreon.com
+ *
+ */
+
+require_once __DIR__ . '/../../class/centreonLog.class.php';
+$centreonLog = new CentreonLog();
+
+//error specific content
+$versionOfTheUpgrade = 'UPGRADE - 21.10.10: ';
+$errorMessage = '';
+
+try {
+    if ($pearDB->isColumnExist('remote_servers', 'server_id') === 0) {
+        $errorMessage = "Unable to add 'server_id' column to remote_servers table";
+        $pearDB->query(
+            "ALTER TABLE remote_servers
+            ADD COLUMN `server_id` int(11) NOT NULL"
+        );
+
+        migrateRemoteServerRelations($pearDB);
+
+        $errorMessage = "Unable to add foreign key constraint of remote_servers.server_id";
+        $pearDB->query(
+            "ALTER TABLE remote_servers
+            ADD CONSTRAINT `remote_server_nagios_server_ibfk_1`
+            FOREIGN KEY(`server_id`) REFERENCES `nagios_server` (`id`)
+            ON DELETE CASCADE"
+        );
+    }
+} catch (\Exception $e) {
+    $centreonLog->insertLog(
+        4,
+        $versionOfTheUpgrade . $errorMessage .
+        " - Code : " . (int)$e->getCode() .
+        " - Error : " . $e->getMessage() .
+        " - Trace : " . $e->getTraceAsString()
+    );
+
+    throw new \Exception($versionOfTheUpgrade . $errorMessage, (int) $e->getCode(), $e);
+}
+
+/**
+ * Manage relations between remote servers and nagios servers
+ *
+ * @param \CentreonDB $pearDB
+ */
+function migrateRemoteServerRelations(\CentreonDB $pearDB): void
+{
+    $processedIps = [];
+
+    $selectServerStatement = $pearDB->prepare(
+        "SELECT id FROM nagios_server WHERE ns_ip_address = :ip_address"
+    );
+    $deleteRemoteStatement = $pearDB->prepare(
+        "DELETE FROM remote_servers WHERE id = :id"
+    );
+    $updateRemoteStatement = $pearDB->prepare(
+        "UPDATE remote_servers SET server_id = :server_id WHERE id = :id"
+    );
+
+    $result = $pearDB->query(
+        "SELECT id, ip FROM remote_servers"
+    );
+    while ($remote = $result->fetch()) {
+        $remoteIp = $remote['ip'];
+        $remoteId = $remote['id'];
+        if (in_array($remoteIp, $processedIps)) {
+            $deleteRemoteStatement->bindValue(':id', $remoteId, \PDO::PARAM_INT);
+            $deleteRemoteStatement->execute();
+        }
+
+        $processedIps[] = $remoteIp;
+
+        $selectServerStatement->bindValue(':ip_address', $remoteIp, \PDO::PARAM_STR);
+        $selectServerStatement->execute();
+        if ($server = $selectServerStatement->fetch()) {
+            $updateRemoteStatement->bindValue(':server_id', $server['id'], \PDO::PARAM_INT);
+            $updateRemoteStatement->bindValue(':id', $remoteId, \PDO::PARAM_INT);
+            $updateRemoteStatement->execute();
+        } else {
+            $deleteRemoteStatement->bindValue(':id', $remoteId, \PDO::PARAM_INT);
+            $deleteRemoteStatement->execute();
+        }
+    }
+}
```

Comment:
```suggestion
function migrateRemoteServerRelations(\CentreonDB $pearDB): void
{
    $processedIps = [];

    $selectStatement = $pearDB->prepare('SELECT id FROM nagios_server WHERE ns_ip_address = :ip_address');
    $deleteStatement = $pearDB->prepare('DELETE FROM remote_servers WHERE id = :id');
    $updateStatement = $pearDB->prepare('UPDATE remote_servers SET server_id = :server_id WHERE id = :id');

    $remoteServers = $pearDB->query('SELECT id, ip FROM remote_servers');

    foreach ($remoteServers->fetchAll(\PDO::FETCH_KEY_PAIR) as $id => $ip) {
        if (in_array($ip, $processedIps)) {
            $deleteStatement->bindValue(':id', $id, \PDO::PARAM_INT);
            $deleteStatement->execute();
        }

        $processedIps[] = $ip;

        $selectStatement->bindValue(':ip_address', $ip, \PDO::PARAM_STR);
        $selectStatement->execute();
        if ($server = $selectStatement->fetch()) {
            $updateStatement->bindValue(':server_id', $server['id'], \PDO::PARAM_INT);
            $updateStatement->bindValue(':id', $id, \PDO::PARAM_INT);
            $updateStatement->execute();
        } else {
            $deleteStatement->bindValue(':id', $id, \PDO::PARAM_INT);
            $deleteStatement->execute();
        }
    }
}
```

Reply:
but you should test if usage of $remoteServers->fetchAll(\PDO::FETCH_KEY_PAIR) is correct